hive/lib/src/sentry_box_base.dart

```diff
@@ -0,0 +1,203 @@
+import 'package:hive/hive.dart';
+import 'package:meta/meta.dart';
+import 'package:sentry/sentry.dart';
+import 'sentry_span_helper.dart';
+
+/// @nodoc
+@internal
+class SentryBoxBase<E> implements BoxBase<E> {
+  final BoxBase<E> _boxBase;
+  final Hub _hub;
+
+  final _spanHelper = SentrySpanHelper(
+    // ignore: invalid_use_of_internal_member
+    SentryTraceOrigins.autoDbHiveBoxBase,
+  );
+
+  /// @nodoc
+  SentryBoxBase(this._boxBase, this._hub) {
+    _spanHelper.setHub(this._hub);
+  }
+
+  @override
+  Future<int> add(E value) async {
+    return _spanHelper.asyncWrapInSpan(
+      'add',
```

Comment:
If possible it'd be great to add more context where it makes sense, so not only `add` and if possible / doesn't expose any sensitive data.

for example a sql span description of: 
`'insert'` vs `'INSERT INTO Product (title) VALUES (?)'` makes quite a difference

Reply:
Difficulty for this particular method, as we only have the method name really. The value is a generic which we cannot infer more info on. Any other methods you had in mind?

Reply:
In that case adding the name of the box should be good as well, afaik it should be accessible through `BoxBase`.

Reply:
kk, we are already adding the box name with every transaction, which we do access from the box base.

```dart
@override
Future<int> add(E value) async {
  return _spanHelper.asyncWrapInSpan(
    'add',
    () async {
      return await _boxBase.add(value);
    },
    dbName: name,
  );
}
```

```dart
@override
String get name => _boxBase.name;
```

Reply:
true, then it's fine 👍